File: JavaScript/Day11.js
(function(){
    "use strict";

    require("fs").readFile("../inputs/Day11.txt", "utf8", function(err, data){
        if(err){
            console.log(`Unable to load file: ${err}`);
        } else {
            let firstPassword = findNextPassword(data);
            let secondPassword = findNextPassword(firstPassword);

            console.log(`Part 1: ${firstPassword}`);
            console.log(`Part 2: ${secondPassword}`);
        }
    });

    let findNextPassword = function(password) {
        let disallowedLetters = /[iol]/,
            doubledLetters = /([a-z])\1.*([a-z])\2/;

        while(true){
            password = (parseInt(password, 36) + 1).toString(36).replace(/0/, 'a');
            if(password.match(doubledLetters) && !password.match(disallowedLetters)){
                let consecutiveLetters = false;
                for(let c = 0; c < password.length - 2; c++){
                    let l1 = password.charCodeAt(c),
                        l2 = password.charCodeAt(c + 1),
                        l3 = password.charCodeAt(c + 2);
                    if(l1 == l2 - 1 && l2 == l3 - 1){
                        return password;
                    }
                }
            }
        }
    };
})();

File: Python/Day11.py
import re
import sys

password = open('../inputs/Day11.txt').read()

def nextPassword(password):
	letters = list(password[::-1])
	for l in range(len(letters)):
		if letters[l] == 'z':
			letters[l] = 'a'
		else:
			letters[l] = chr(ord(letters[l]) + 1)
			break
	return ''.join(letters[::-1])

def validPassword(password):
	if re.search('[iol]', password) == None and len(re.findall(r'([a-z])\1', password)) >= 2:
		for c in range(len(password) - 2):
			if ord(password[c]) == ord(password[c + 1]) - 1 and ord(password[c + 1]) == ord(password[c + 2]) - 1:
				return True
	return False

while True:
	password = nextPassword(password)
	if validPassword(password):
		break

firstPassword = password

while True:
	password = nextPassword(password)
	if validPassword(password):
		break

secondPassword = password

print "Part 1:", firstPassword
print "Part 2:", secondPassword